src/components/modal-dialog/modal-dialog.tsx

```diff
@@ -0,0 +1,118 @@
+import styled, { useTheme } from "styled-components";
+import { ActionButton } from "../action-button/action-button";
+import { ActionButtonVariant } from "../../types";
+import CloseIcon from "../../../public/icons/close.svg";
+
+const Overlay = styled.div`
+  position: fixed;
+  left: 0;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background: #00000099;
+  z-index: 103;
+`;
+
+const Container = styled.div`
+  position: absolute;
+  display: flex;
+  flex-direction: column;
+  border-radius: 8px;
+  background: ${({ theme }) => theme.colors.mainColor};
+  z-index: 104;
+`;
+
+const IconContainer = styled.div`
+  position: absolute;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border-radius: 8px;
+  top: 13px;
+  right: 14px;
+  width: 44px;
+  height: 44px;
+`;
+
+const ContentContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: start;
+  height: 100%;
+  width: 100%;
+  margin: 32px 32px 0 32px;
+  row-gap: 16px;
+  color: ${({ theme }) => theme.colors.fontColor};
+`;
+
+const Title = styled.div`
+  font-style: normal;
+  font-weight: 700;
+  font-size: 32px;
+  line-height: 45px;
+`;
+
+const ButtonsContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  margin: 32px;
+  column-gap: 18px;
+   {
+    > * {
+      width: 180px;
+    }
+  }
+`;
+
+type LogoutPanelProps = {
+  title: string;
+  content: (() => JSX.Element) | JSX.Element;
+  okButtonText?: string;
+  cancelButtonText?: string;
+  onCancel: () => void;
+  onConfirm: () => void;
+};
+
+const CloseCrossButton = styled(CloseIcon)`
+  &:hover {
+    fill: ${({ theme }) => theme.colors.mainDimColorInverted};
+  }
+`;
+
+export const ModalDialog = ({
+  title,
+  content,
+  cancelButtonText = "Cancel",
+  okButtonText = "Ok",
+  onCancel,
+  onConfirm,
+}: LogoutPanelProps) => {
+  const theme = useTheme();
+
+  return (
+    <Overlay>
```

Comment:
`Overlay` trasparency is applied on child components. We cannot use this approach. Instead, we should create another `WrapperContainer` with `position: fixed` and so on with `visibility: hidden`. We set `visibility: visible` for the child and it will be shown in the center.

Reply:
The version that is submitted uses this approach, but actually it doesn't work. The problem is that the "ModalDialog" component is a child of some other component that uses opaque. It's out of my control.
I think we have to use the Popover that renders the dialog in the body, out of any other components.